mypy_django_plugin/transformers/fields.py

```diff
@@ -201,6 +205,156 @@ def drop_combinable(_type: MypyType) -> Optional[MypyType]:
     return helpers.reparametrize_instance(default_return_type, args)
 
 
+class M2MThrough(NamedTuple):
+    arg: Optional[Expression]
+    model: ProperType
+
+
+class M2MTo(NamedTuple):
+    arg: Expression
+    model: ProperType
+    self: bool  # ManyToManyField('self', ...)
+
+
+class M2MArguments(NamedTuple):
+    to: M2MTo
+    through: Optional[M2MThrough]
+
+
+def get_model_from_expression(
+    expr: Expression,
+    *,
+    api: Union[TypeChecker, SemanticAnalyzer],
+    django_context: DjangoContext,
+) -> Optional[ProperType]:
+    """
+    Attempts to resolve an expression to a 'TypeInfo' instance. Any lazy reference
+    argument(e.g. "<app_label>.<object_name>") to a Django model is also attempted.
+    """
+    # TODO: Handle settings.AUTH_USER_MODEL?
+    if isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):
+        if (
+            expr.node.metaclass_type is not None
+            and expr.node.metaclass_type.type.fullname == fullnames.MODEL_METACLASS_FULLNAME
+        ):
+            return Instance(expr.node, [])
+    elif isinstance(expr, StrExpr):
+        if "." not in expr.value:
+            # <object_name> -- needs prefix of <app_label>. We can't implicitly solve
+            # what app label this should be, yet.
+            api.fail("Could not resolve lazy reference without an app label", expr)
+            api.note(
+                ("Try to use a reference explicitly prefixed with app label:" f' "<app_label>.{expr.value}" instead'),
+                expr,
+            )
+            return None
+
+        # Expression conforms to the structure of a lazy reference: '<app_label>.<object_name>'
+        fullname = django_context.model_class_fullnames_by_label.get(expr.value)
+        if fullname is not None:
```

Comment:
Hm, I think we could yield an error on the `else` clause here, saying that the lazy reference is invalid and doesn't point to a model